storage/database/db_manager.go

```diff
@@ -2697,6 +2689,30 @@ func (dbm *databaseManager) WriteGovernanceIdx(num uint64) error {
 	return db.Put(governanceHistoryKey, data)
 }
 
+// DeleteGovernanceIdxRange leaves governanceHistory from 0 to `num`
```

Comment:
```suggestion
// DeleteGovernanceIdxRange deletes governanceHistory beyond `num`
```

Reply:
But since `delFn` and `DeleteGovernance` is called at every block,
This function might peek at the last element of the idxHistory and delete on exact match.

Reply:
The governance index is read and stored in the unit of the list by the existing implementation. So, the function `deleteGovernanceIdxRange` stores an index list from 0 to `num`. To peek at one element, the existing implementation should be rewritten (not changed in this PR) at read and store by a pair of number and index.

Reply:
Oh, by peeking at the last element, I simply meant something like
```go
end := len(idxHistory)
if idxHistory[end-1] == num {
  data, err := json.Marshal(idxHistory[0:end-1])
  db.Put(governanceHistoryKey, data)
}
```
No need to change the database schema.

Reply:
Do you mean the change provided can make a shortcut, which is an alternative to the following loop?

```go
for i := 0; i < len(idxHistory); i++ {
    if idxHistory[i] > num {
        break
    }
    newSlice = append(newSlice, idxHistory[i])
}
```

Reply:
I wanted to make `DeleteGovernance` analogous to `DeleteBody` and `DeleteReceipts` in that they delete the elements strictly at `num`.

Your loop indeed handles every element at once and is therefore efficient, if called only once.
But `DeleteGovernance` is called multiple times in `bc.hc.SetHead`, because of the loop that calls `delFn` for every block number. The time complexity here is `len(idxHistory) * [number of blocks rewinded]`.
That's why I suggested making `DeleteGovernanceIdxRange` an O(1) algorithm without the loop.

Reply:
In case the suggestion is accepted, the name should be revised as well: maybe `deleteGovernanceIdx`

Reply:
Changed. If the function name feels verbose or not appropriate, please suggest an alternative one.